presto-main/src/main/java/com/facebook/presto/type/ArrayType.java

```diff
@@ -76,8 +76,12 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
         }
 
         for (int i = 0; i < leftArray.getPositionCount(); i++) {
-            checkElementNotNull(leftArray.isNull(i), ARRAY_NULL_ELEMENT_MSG);
-            checkElementNotNull(rightArray.isNull(i), ARRAY_NULL_ELEMENT_MSG);
+            if (leftArray.isNull(i) || rightArray.isNull(i)) {
+                if (leftArray.isNull(i) ^ rightArray.isNull(i)) {
```

Comment:
I feel `!=` is more proper here than `^` because we are not trying to do a `bitwise` operation here. But I'm ok either way if you feel strongly about using `^`. Ditto for the other two occurrences.